### PR TITLE
ES-647590-CommonMAUIFile

### PR DESCRIPTION
**Pull request by Kiruthika-3470**
Hi @Keerthivasan-Ramamoorthy  sir,

Implement common file sharing for mind map and flow chart samples in maui application. Can you please check this code changes?

Regards,
Kiruthika M

### CODE REVIEW

File: Common/Pages/DiagramMenuBar.razor [hunk 1]
```diff
@@ -440,8 +455,18 @@
                 await SaveDialogBox.SaveDialog.Show();
                 break;
             case "print":
+                #if SERVER
                 File = true;
                 await PrintDialogBox.PrintDialog.ShowAsync();
+                #endif
+                #if WASM
+                File = true;
```
Comment (Keerthivasan-Ramamoorthy) [inline Common/Pages/DiagramMenuBar.razor:463]: Add or condition and move to common code for both server and wasm

File: Common/Pages/DiagramOpenDialog.razor [hunk 2]
```diff
@@ -213,6 +213,24 @@
...
+        TemplateType = "Template1";
+        await Parent.Toolbar.HideElements("hide-palette");
+        Parent.MindMapPropertyPanel.bindMindMapProperties();
+        RemoveUnwantedMindmapMenubarItems();
+        IsOpenDialogVisible = false;
+        this.Parent.DiagramContent.show = false;
+        StateHasChanged();
+        Parent.LeftSideBar.sidebarVisibility = "none";
+        Parent.LeftSideBar.StateChanged();
+        #endif
```
Comment (Keerthivasan-Ramamoorthy) [inline Common/Pages/DiagramOpenDialog.razor:233]: Move common code to outside of the server, wasm and MAUI condtions.

File: Common/Pages/DiagramOpenDialog.razor [hunk 3]
```diff
@@ -255,6 +273,24 @@
...
+    this.Parent.DiagramContent.show = false;
+    StateHasChanged();
+    CurrentDiagramVisibility("mindmap-diagram");
+    await Parent.MenuBar.BusinessPlanningDiagram();
+    TemplateType = "Template2";
+    Parent.MindMapPropertyPanel.bindMindMapProperties();
+    RemoveUnwantedMindmapMenubarItems();
+    Parent.LeftSideBar.sidebarVisibility = "none";
+    Parent.LeftSideBar.StateChanged();
+    #endif
```
Comment (Keerthivasan-Ramamoorthy) [inline Common/Pages/DiagramOpenDialog.razor:293]: Move common code to outside of the server, wasm and MAUI condtions.

File: Common/Pages/DiagramOpenDialog.razor [hunk 4]
```diff
@@ -295,7 +331,25 @@
...
+    this.Parent.DiagramContent.show = false;
+    StateHasChanged();
+    CurrentDiagramVisibility("mindmap-diagram");
+    await Parent.MenuBar.TQMDiagram();
+    TemplateType = "Template3";
+    Parent.MindMapPropertyPanel.bindMindMapProperties();
+    RemoveUnwantedMindmapMenubarItems();
+    Parent.LeftSideBar.sidebarVisibility = "none";
+    Parent.LeftSideBar.StateChanged();
+    #endif
```
Comment (Keerthivasan-Ramamoorthy) [inline Common/Pages/DiagramOpenDialog.razor:352]: Move common code to outside of the server, wasm and MAUI condtions.

File: Common/Pages/DiagramOpenDialog.razor [hunk 5]
```diff
@@ -337,6 +391,24 @@
...
+    this.Parent.DiagramContent.show = false;
+    StateHasChanged();
+    CurrentDiagramVisibility("mindmap-diagram");
+    await Parent.MenuBar.SoftwareLifeCycleDiagram();
+    TemplateType = "Template4";
+    Parent.MindMapPropertyPanel.bindMindMapProperties();
+    RemoveUnwantedMindmapMenubarItems();
+    Parent.LeftSideBar.sidebarVisibility = "none";
+    Parent.LeftSideBar.StateChanged();
+    #endif
```
Comment (Keerthivasan-Ramamoorthy) [inline Common/Pages/DiagramOpenDialog.razor:411]: Move common code to outside of the server, wasm and MAUI condtions.

**Review by Keerthivasan-Ramamoorthy — APPROVED** [review]

File: Common/Pages/DiagramOpenDialog.razor (hunk 3 above)
Comment (Keerthivasan-Ramamoorthy) [reply Common/Pages/DiagramOpenDialog.razor]: resolved

**Review by sarathkumarvenkatesan — APPROVED** [review]

**Pull request merged**